### PR TITLE
adding onlyText button variant

### PR DESCRIPTION
**Pull request by npfitz**
Adding text only button variant. No borders or padding around the button content, and color defaults to brand.

image

### CODE REVIEW

File: frontend/src/metabase/core/components/Button/Button.styled.tsx [hunk 1]
```diff
@@ -23,6 +24,21 @@ export const ButtonRoot = styled.button<ButtonRootProps>`
...
       }
     `}
+
+  ${({ onlyText }) =>
+    onlyText &&
+    css`
+      border: none;
+      padding: 0;
+
+      ${ButtonContent} {
```
Comment (ranquild) [inline frontend/src/metabase/core/components/Button/Button.styled.tsx:34]: Could we simply set `color` on the top level component without targeting `ButtonContent`? It seems that `purple` variant does that.

Comment (npfitz) [reply frontend/src/metabase/core/components/Button/Button.styled.tsx:34]: Certainly can!

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/core/components/Button/Button.styled.tsx [hunk 2]
```diff
@@ -23,6 +24,18 @@ export const ButtonRoot = styled.button<ButtonRootProps>`
...
+
+  ${({ onlyText }) =>
+    onlyText &&
+    css`
+      border: none;
+      padding: 0;
+
+      color: ${color("brand")};
+      &:hover {
+        background-color: unset;
```
[the remark this answers is not in the document]
Comment (iethree) [reply frontend/src/metabase/core/components/Button/Button.styled.tsx:36]: too slow...

[the remark this answers is not in the document]
Comment (npfitz) [reply frontend/src/metabase/core/components/Button/Button.styled.tsx:36]: We can still update the default behavior for it. I didn't think we actually wanted a hover state, but it's possible we do

**Pull request merged**